grpc/server/src/engine_server/mappings/state/auction_info.rs

```diff
@@ -0,0 +1,107 @@
+                delegator.set_delegator_public_key(delegator_public_key_bytes);
+                delegator.set_validator_public_key(validator_public_key_bytes);
+                delegator.set_amount(amount);
+                ret.set_delegator(delegator)
+            }
+        }
+        ret
+    }
+}
+
+impl TryFrom<state::SeigniorageAllocation> for SeigniorageAllocation {
+    type Error = ParsingError;
+
+    fn try_from(
+        pb_seigniorage_allocation: state::SeigniorageAllocation,
+    ) -> Result<Self, Self::Error> {
+        let pb_variants = pb_seigniorage_allocation.variants.as_ref().ok_or_else(|| {
+            ParsingError("Unable to parse Protobuf SeigniorageAllocation".to_string())
+        })?;
+
+        match pb_variants {
+            SeigniorageAllocation_oneof_variants::validator(pb_validator) => {
+                let validator_public_key =
+                    bytesrepr::deserialize(pb_validator.get_validator_public_key().to_vec())?;
+                let amount = U512::try_from(pb_validator.get_amount().to_owned())?;
+                Ok(SeigniorageAllocation::validator(
+                    validator_public_key,
+                    amount,
+                ))
+            }
+            SeigniorageAllocation_oneof_variants::delegator(pb_delegator) => {
+                let delegator_public_key =
+                    bytesrepr::deserialize(pb_delegator.get_delegator_public_key().to_vec())?;
+                let validator_public_key =
+                    bytesrepr::deserialize(pb_delegator.get_validator_public_key().to_vec())?;
+                let amount = U512::try_from(pb_delegator.get_amount().to_owned())?;
+                Ok(SeigniorageAllocation::delegator(
+                    delegator_public_key,
+                    validator_public_key,
+                    amount,
+                ))
+            }
+        }
+    }
+}
+
+impl From<AuctionInfo> for state::AuctionInfo {
+    fn from(auction_info: AuctionInfo) -> Self {
+        let mut ret = state::AuctionInfo::new();
+        let mut pb_vec_seigniorage_allocations = Vec::new();
+        for allocation in auction_info.seigniorage_allocations().iter() {
+            pb_vec_seigniorage_allocations.push(allocation.to_owned().into());
+        }
+        ret.set_seigniorage_allocations(pb_vec_seigniorage_allocations.into());
+        ret
+    }
+}
+
+impl TryFrom<state::AuctionInfo> for AuctionInfo {
+    type Error = ParsingError;
+
+    fn try_from(pb_auction_info: state::AuctionInfo) -> Result<Self, Self::Error> {
+        let mut seigniorage_allocations = Vec::new();
+        for pb_seigniorage_allocation in pb_auction_info.get_seigniorage_allocations().iter() {
+            seigniorage_allocations.push(pb_seigniorage_allocation.to_owned().try_into()?);
+        }
+        let mut ret = AuctionInfo::new();
+        *ret.seigniorage_allocations_mut() = seigniorage_allocations;
+        Ok(ret)
+    }
+}
```

Comment:
Since you have already generators defined for AuctionInfo it'd be worth to add a test. There's a lot of boilerplate in mappings module to copy from i.e.:

```rust
#[cfg(test)]
mod tests {
    proptest! {
        #[test]
        fn round_trip(auction_info in gens::auction_info_arb()) {
            test_utils::protobuf_round_trip::<AuctionInfo, ipc::AuctionInfo>(auction_info );
        }
    }
}
```

Reply:
This proved to be a little more complicated than expected because of the module structure and access modifiers on the various `gens` module.  If we are going to do, we should also do for `DeployInfo` and `Transfer` as well, so I will ticket some tech debt to add this.  Right now the code is getting exercised by the test suites, so I am pretty confident in these mappings, but I agree that it would be nice have a little extra confidence.